### PR TITLE
Cleanup TimerSchedulerImpl & TimerInterceptor

### PR DESCRIPTION
**Pull request by dwijnand**

### CODE REVIEW

File: akka-actor-typed/src/main/scala/akka/actor/typed/internal/TimerSchedulerImpl.scala [hunk 1]
```diff
@@ -75,9 +67,9 @@ import scala.concurrent.duration.FiniteDuration
 
     val timerMsg =
       if (msg.isInstanceOf[NotInfluenceReceiveTimeout])
-        NotInfluenceReceiveTimeoutTimerMsg(key, nextGen, this)
+        new TimerMsg(key, nextGen, this) with NotInfluenceReceiveTimeout
```
Comment (patriknw) [inline akka-actor-typed/src/main/scala/akka/actor/typed/internal/TimerSchedulerImpl.scala]: `case class` should preferably be `final` according to normal Scala conventions, so perhaps make it ordinary `class` and implement `toString`

**Review by patriknw — COMMENTED** [review]
Great cleanup

**Review by patriknw — APPROVED** [review]
LGTM, thanks

**Comment by dwijnand** [comment]
Can this land now?

**Review by chbatey — APPROVED** [review]
LGTM

**Pull request merged**